src/classes/AFFL_MultiRecordType_TDTM.cls

```diff
@@ -141,7 +141,7 @@ public class AFFL_MultiRecordType_TDTM extends TDTM_Runnable {
 					Contact relatedContact = relatedContactsMap.get(afflOld.Contact__c);
 
 					//AFTER DELETE - delete lookup relationship, if necessary
-					if (triggerAction == TDTM_Runnable.Action.AfterDelete && lookupFieldName != null) {
+					if (triggerAction == TDTM_Runnable.Action.AfterDelete && lookupFieldName != null && relatedContact != null) {
```

Comment:
Especially because the processAfflDeleted() method is a public method, I would prefer we do the null check in the method itself rather than before calling the method (or in addition to)